### PR TITLE
FQLoRA: use last checkpoint with AWQ+SE as initialization

### PR DESCRIPTION
**Pull request by ljaljushkin**
### Changes

- Fixed support for more advanced compression methods (AWQ+Scale Estimation) as initialization for QAT with absorbable LoRA. 
Previously RTN baseline was used, but now with data-aware AWQ + Scale Estimation tuning starts with more accurate model and achieves superior accuracy: 

image

- Removed selection of the best checkpoint based on validation set. It significantly reduces overall tuning time and max allocated memory. The best results are slightly worse on wikitext, but it should provide more fair and faster tuning pipeline (from 32 min to 25 min for SmoLM-1.7B). 

### Reason for changes

Improvements in QAT + absorbable LoRA.

### Related tickets

ticket - 169140

### Tests

test_examples - https://github.com/openvinotoolkit/nncf/actions/runs/16223789950

### CODE REVIEW

File: examples/llm_compression/torch/distillation_qat_with_lora/README.md [hunk 1]
```diff
@@ -55,37 +55,37 @@ Where:
...
-| mistralai/Mistral-7B-v0.3           | INT4 (QAT + LoRA) | 8.36                  | 21%         |
+| mistralai/Mistral-7B-v0.3           | INT4 (QAT + LoRA) | 8.35                  | 23%         |
 | mistralai/Mistral-7B-v0.3           | INT4 (best PTWC)  | 8.40                  |             |
-| Qwen/Qwen2.5-3B-Instruct            | BF16              | 11.01                 |             |
-| Qwen/Qwen2.5-3B-Instruct            | INT4 (QAT + LoRA) | 11.45                 | 29%         |
+| Qwen/Qwen2.5-3B-Instruct            | BF16              | 11.02                 |             |
+| Qwen/Qwen2.5-3B-Instruct            | INT4 (QAT + LoRA) | 11.48                 | 27%         |
 | Qwen/Qwen2.5-3B-Instruct            | INT4 (best PTWC)  | 11.64                 |             |
-|                                     |                   |               Average | 46%         |
+|                                     |                   |               Average | 39%         |
```
Comment (nikita-savelyevv) [inline examples/llm_compression/torch/distillation_qat_with_lora/README.md:91]: I see that despite initializing with SE & AWQ, average improvement has decreased. Is this because of best checkpoint selection removal?

File: examples/llm_compression/torch/distillation_qat_with_lora/main.py [hunk 2]
```diff
@@ -359,53 +325,52 @@ def main(argv) -> float:
...
         group_size=64,
+        awq=True,
+        scale_estimation=True,
         compression_format=CompressionFormat.FQ_LORA,
-        advanced_parameters=AdvancedCompressionParameters(lora_adapter_rank=args.lora_rank),
+        advanced_parameters=AdvancedCompressionParameters(
+            awq_params=AdvancedAWQParameters(prefer_data_aware_scaling=True), lora_adapter_rank=args.lora_rank
+        ),
     )
+    pprint(compression_config)
```
Comment (nikita-savelyevv) [inline examples/llm_compression/torch/distillation_qat_with_lora/main.py]: Is this intended? If so, perhaps some preliminary description should be added like "Compression config:"?

File: examples/llm_compression/torch/distillation_qat_with_lora/main.py [hunk 3]
```diff
@@ -359,53 +325,52 @@ def main(argv) -> float:
...
     else:
-        model = compress_weights(model, dataset=Dataset([example_input]), **compression_config)
+        model = compress_weights(model, dataset=calib_dataset, **compression_config)
         save_checkpoint(model, ckpt_file)
     fq_lr = args.lr / 10
     weight_decay = args.lr
     param_to_train = set_trainable(model, lora_lr=args.lr, fq_lr=fq_lr)
     opt = torch.optim.AdamW(param_to_train, weight_decay=weight_decay)
 
-    with create_eval_model(model, args.fast_eval, args.pretrained, torch_dtype, ckpt_file) as eval_model:
```
Comment (nikita-savelyevv) [inline examples/llm_compression/torch/distillation_qat_with_lora/main.py:402]: I believe `fast_eval` should now be removed from CLI arguments.

File: examples/llm_compression/torch/distillation_qat_with_lora/main.py [hunk 4]
```diff
@@ -76,64 +75,24 @@ def get_wikitext2(num_samples: int, seqlen: int, tokenizer: Any, device: torch.d
...
-    automatically deleted when the context exits. Otherwise, uses the provided model.
-
-    :param model: Original model to use if fast_eval is False.
-    :param fast_eval: Whether to create a new optimized model for evaluation.
-    :param pretrained: Pretrained model identifier or path for AutoModelForCausalLM.
-    :param torch_dtype: PyTorch data type to use for the model (e.g., torch.bfloat16).
-    :param ckpt_file: Path to the checkpoint file to load weights from.
-    :yields: Model to use for evaluation, either the new loaded model or the given one.
-    """
-    if fast_eval:
```
Comment (nikita-savelyevv) [inline examples/llm_compression/torch/distillation_qat_with_lora/main.py:100]: Have you tried to combine fast eval approach with AWQ? I wonder if removal of fast eval is required for the updated example to work.

File: examples/llm_compression/torch/distillation_qat_with_lora/README.md (hunk 1 above)
Comment (ljaljushkin) [reply examples/llm_compression/torch/distillation_qat_with_lora/README.md:91]: yes

File: examples/llm_compression/torch/distillation_qat_with_lora/main.py (hunk 4 above)
Comment (ljaljushkin) [reply examples/llm_compression/torch/distillation_qat_with_lora/main.py:100]: without eval it's more fair and could avoid overfit to a small val tass. many relevant works take the last checkpoint

File: examples/llm_compression/torch/distillation_qat_with_lora/main.py (hunk 3 above)
Comment (ljaljushkin) [reply examples/llm_compression/torch/distillation_qat_with_lora/main.py:402]: removed

File: examples/llm_compression/torch/distillation_qat_with_lora/main.py (hunk 2 above)
Comment (ljaljushkin) [reply examples/llm_compression/torch/distillation_qat_with_lora/main.py]: done

File: examples/llm_compression/torch/distillation_qat_with_lora/README.md (hunk 1 above)
Comment (nikita-savelyevv) [reply examples/llm_compression/torch/distillation_qat_with_lora/README.md:91]: My concern is that when we do two changes at the same time (addition for SE/AWQ + removal of best-selection-logic) it's hard to judge contribution of each change to the resulting metrics. Have you measured these changes separately, e.g. data-aware initialization with keeping best checkpoint saving or data-free initialization without best checkpoint saving?

Comment (ljaljushkin) [reply examples/llm_compression/torch/distillation_qat_with_lora/README.md:91]: `data-free initialization without best checkpoint` in the table. SE/AWQ do gives better results in the same setup

File: examples/llm_compression/torch/downstream_qat_with_nls/main.py [hunk 5]
```diff
@@ -95,15 +95,20 @@ def set_trainable(model: nn.Module, lora_lr: float, fq_lr: float) -> list[dict[s
     return [{"params": adapters_to_train, "lr": lora_lr}, {"params": scales_to_train, "lr": fq_lr}]
 
 
-def save_checkpoint(model: nn.Module, ckpt_file: Path) -> None:
+def save_checkpoint(model: nn.Module, ckpt_file: Path, model_state: bool = True) -> None:
```
Comment (andreyanufr) [inline examples/llm_compression/torch/downstream_qat_with_nls/main.py:98]: Why we need model_state=True, if it is always False in this example?

Comment (ljaljushkin) [reply examples/llm_compression/torch/downstream_qat_with_nls/main.py:98]: It should be False in the end of each epoch, corrected.
also added --basic_init CLI argument to handle both cases in the sample.
